calico/felix/test/test_config.py

```diff
@@ -273,4 +283,115 @@ def test_ip_in_ip_enabled_bad(self):
         with self.assertRaisesRegexp(ConfigException,
                                      "Field was not a valid Boolean"
                                      ".*IpInIpEnabled"):
-            config.report_etcd_config({}, cfg_dict)
\ No newline at end of file
+            config.report_etcd_config({}, cfg_dict)
+
+    def test_reporting_ttl_not_int(self):
+        """
+        Test exception is rased if status report ttl has invalid (non-integer) value.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingTTLSecs": "NaN"}
+        with self.assertRaisesRegexp(ConfigException,
+                                     "Field was not integer.*"):
+            config.report_etcd_config({}, cfg_dict)
+
+    def test_reporting_interval_not_int(self):
+        """
+        Test exception is rased ifstatus reporting interval has invalid (non-integer) value.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingIntervalSecs": "NaN"}
+        with self.assertRaisesRegexp(ConfigException,
+                                     "Field was not integer.*"):
```

Comment:
Ditto above comment about your more specific error checking code in config.py either not being necessary or this test isn't working correctly.